<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <link rel="icon" href="../favicon.ico" />
  <link rel="stylesheet" href="../style.css">
  <title>Ruby を使用した postgres でクエリが失敗したかどうかを確認するにはどうすればよいですか? | Techioz</title>
  <style type="text/css">code{white-space: pre;}</style>
  <script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-1637770322917058"
     crossorigin="anonymous"></script>
  <!-- Google tag (gtag.js) -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-DRM1ZRNLDN"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-DRM1ZRNLDN');
  </script>
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Article",
  "headline": "Ruby を使用した postgres でクエリが失敗したかどうかを確認するにはどうすればよいですか?",
  "author": {
    "@type": "Person",
    "address": "",
    "email": null,
    "identifier": "masyumaroking",
    "name": "masyumaroking"
  },
  "publisher": {
    "@type": "Person",
    "name": "masyumaroking"
  }
}
</script>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
  <header>
    <div class="container">
        <h1>Techioz Blog</h1>
        <nav>
            <ul>
                <li><a href="#">Home</a></li>
                <li><a href="#">Articles</a></li>
                <li><a href="#">About</a></li>
                <li><a href="#">Contact</a></li>
            </ul>
        </nav>
    </div>
  </header>
  <main>
    <div class="container">
        <article>
                      <h2>Ruby を使用した postgres
でクエリが失敗したかどうかを確認するにはどうすればよいですか?</h2>
                    <h2 id="概要">概要</h2>
                    <p>これはコンテキストの基本的なコードです</p>
                    <pre><code>require &#39;pg&#39;

conn = PG.connect(db_params)
# example query:
response = conn.exec(&quot;update employees set id = 0 where role = &#39;owner&#39;&quot;)

puts response
</code></pre>
                    <p>select
                    ステートメントではないステートメントの応答を出力しようとするたびに、前のコードが示すように、クエリの実行時に作成される
                    Result オブジェクトのアドレスを取得します。</p>
                    <p>クエリが失敗したかどうかを知る方法はありますか?</p>
                    <p>「pg」gem
                    ドキュメントにアクセスしましたが、select
                    ステートメントの実行についてのみ言及されており、私の場合には役に立ちませんでした。</p>
                    <h2 id="解決策">解決策</h2>
                    <p>PG::Connection#exec
                    のドキュメントには次のように書かれています。</p>
                    <p>したがって、クエリが失敗すると、PG::Error
                    タイプの例外が発生します。 begin …rescue
                    ブロックを使用してそれをキャッチできます。</p>
                    <pre><code>begin
  conn.exec(&quot;update employees set id = &#39;invalid&#39; where role = &#39;owner&#39;&quot;)
rescue PG::Error =&gt; e
  puts &quot;#{e.class}: #{e.message}&quot;
end
</code></pre>
        </article>
    </div>
  </main>
  <footer>
    <div class="container">
        <p>&copy; 2024 Techioz. All rights reserved.</p>
    </div>
  </footer>
</body>
</html>